app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEActivity.java

```diff
@@ -423,7 +423,10 @@ private void showMap(boolean showMap) {
             }
 
             if (getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT) {
-                setFabVisibility(!needsSearch.get() && !showMap, true);
+                setFabVisibility(
+                        backDropActive || (!needsSearch.get() && !showMap),
+                        !backDropActive
```

Comment:
I would extract this to a method splitting responsibilities.
`setFabVisibility(isButtonVisible())`